examples/structured_2d_dgsem/elixir_eulerpolytropic_coupled.jl

```diff
@@ -0,0 +1,139 @@
+
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# Couple polytropic Euler systems.
+#
+# In a rectangular domain we solve two different sets of equations for the
+# left and the right half. Those are the isothermal equations (left)
+# and polytropic equations (right).
+# The coupling hapens on two interfaces. One is located at the center of the
+# domain such that the right half is coupled through its left boundary
+# and the left half is coupled through its right boundary. The other coupling
+# makes sure that the domain is periodic. Here the right boundary of the right
+# domain is coupled to the left boundary of the left domain.
+# The vertical boundaries are simply periodic.
+# As test case we send a linear wave through the domain and observe a change
+# in the dispersion relation when the wave enters the isothermal domain.
+
+###############################################################################
+# define the initial conditions
+
+function initial_condition_wave_isothermal(x, t, equations::PolytropicEulerEquations2D)
+    gamma = 1.0
+    kappa = 1.0
+
+    rho = 1.0
+    v1 = 0.0
+    v2 = 0.0
+
+    return prim2cons(SVector(rho, v1, v2), equations)
+end
+
+function initial_condition_wave_polytropic(x, t, equations::PolytropicEulerEquations2D)
+    gamma = 2.0
+    kappa = 1.0
```

Comment:
Could you also just use `equations.gamma` and `equations.kappa`? Same holds for other initial conditions.

Reply:
Done!